diff -ruN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2026-08-06 12:47:18
+++ b/CMakeLists.txt	2026-08-06 12:47:19
@@ -63,6 +63,11 @@
     set(make_env ${CMAKE_COMMAND} -E env SDKROOT=${CMAKE_OSX_SYSROOT})
 endif()
 
+option(ABC_USE_SYSTEM_ZLIB "Use system shared ZLIB" OFF)
+if(ABC_USE_SYSTEM_ZLIB)
+    set(ABC_USE_SYSTEM_ZLIB_FLAGS "ABC_USE_SYSTEM_ZLIB=${ABC_USE_SYSTEM_ZLIB}")
+endif()
+
 # run make to extract compiler options, linker options and list of source files
 execute_process(
   COMMAND
@@ -71,6 +76,7 @@
         ${ABC_READLINE_FLAGS}
         ${ABC_USE_NAMESPACE_FLAGS}
         ${ABC_USE_STDINT_H_FLAGS}
+        ${ABC_USE_SYSTEM_ZLIB_FLAGS}
         ARCHFLAGS_EXE=${CMAKE_CURRENT_BINARY_DIR}/abc_arch_flags_program.exe
         ABC_MAKE_NO_DEPS=1
         CC=${CMAKE_C_COMPILER}
@@ -100,10 +106,16 @@
     set_source_files_properties(${ABC_SRC} PROPERTIES LANGUAGE CXX)
 endif()
 
+
+find_package(ZLIB)
+
 function(abc_properties target visibility)
     target_include_directories(${target} ${visibility} ${CMAKE_CURRENT_SOURCE_DIR}/src )
     target_compile_options_filtered(${target} ${visibility} ${ABC_CFLAGS} ${ABC_CXXFLAGS} -Wno-unused-but-set-variable )
     target_link_libraries(${target} ${visibility} ${ABC_LIBS})
+    if(ABC_USE_SYSTEM_ZLIB)
+        target_link_libraries(${target} ${visibility} ZLIB::ZLIB)
+    endif()
 endfunction()
 
 set(ABC_MAIN_SRC src/base/main/main.c)
diff -ruN a/Makefile b/Makefile
--- a/Makefile	2026-08-06 12:47:18
+++ b/Makefile	2026-08-06 12:47:28
@@ -39,7 +39,7 @@
 	src/map/mapper src/map/mio src/map/super src/map/if src/map/if/acd \
 	src/map/amap src/map/cov src/map/scl src/map/mpm src/map/emap \
 	src/misc/extra src/misc/mvc src/misc/st src/misc/util src/misc/nm \
-	src/misc/vec src/misc/hash src/misc/tim src/misc/bzlib src/misc/zlib \
+	src/misc/vec src/misc/hash src/misc/tim src/misc/bzlib \
 	src/misc/mem src/misc/bar src/misc/bbl src/misc/parse src/misc/btor \
 	src/opt/cut src/opt/fxu src/opt/fxch src/opt/rwr src/opt/mfs src/opt/sim \
 	src/opt/ret src/opt/fret src/opt/res src/opt/lpk src/opt/nwk src/opt/rwt src/opt/rar \
@@ -61,6 +61,12 @@
 	$(CC) $< -o $(ARCHFLAGS_EXE)
 
 INCLUDES += -I$(ABCSRC)/src
+
+ifdef ABC_USE_SYSTEM_ZLIB
+	CFLAGS += -DUSE_SYSTEM_ZLIB
+else
+	MODULES += src/misc/zlib
+endif
 
 # Use C99 stdint.h header for platform-dependent types
 ifdef ABC_USE_STDINT_H
diff -ruN a/src/misc/zlib/zlib.h b/src/misc/zlib/zlib.h
--- a/src/misc/zlib/zlib.h	2026-08-06 12:47:18
+++ b/src/misc/zlib/zlib.h	2026-08-06 12:47:19
@@ -28,6 +28,10 @@
   (zlib format), rfc1951 (deflate format) and rfc1952 (gzip format).
 */
 
+#ifdef USE_SYSTEM_ZLIB
+#include "misc/util/abc_global.h"
+#include <zlib.h>
+#else
 #ifndef ZLIB_H
 #define ZLIB_H
 
@@ -2057,3 +2061,4 @@
 ABC_NAMESPACE_HEADER_END
 
 #endif /* ZLIB_H */
+#endif /* USE_SYSTEM_ZLIB */
